manage_breast_screening/auth/views.py

```diff
@@ -78,3 +86,65 @@ def jwks(request):
         return JsonResponse({"keys": [jwk_dict]})
     except Exception:
         return JsonResponse({"keys": []})
+
+
+@require_http_methods(["POST"])
+@csrf_exempt
+@login_not_required
+def cis2_back_channel_logout(request):
+    """Handle CIS2 back-channel logout notifications.
+
+    This endpoint receives logout tokens from CIS2 when a user's session is terminated.
+    See: https://openid.net/specs/openid-connect-backchannel-1_0.html
+    See: https://digital.nhs.uk/services/care-identity-service/applications-and-services/cis2-authentication/guidance-for-developers/detailed-guidance/session-management#back-channel-logout
+    """
+    logout_token = request.POST.get("logout_token")
+    if not logout_token:
+        return HttpResponseBadRequest("Missing logout_token")
+
+    # Get the CIS2 client and prepare key loader for token verification
+    client = get_cis2_client()
+    metadata = client.load_server_metadata()
+    key_loader = client.create_load_key()
+    try:
+        verification_rules = {
+            "iss": {"values": [metadata["issuer"]], "essential": True},
+            "aud": {"values": [client.client_id], "essential": True},
+            "exp": {"essential": True},
```

Comment:
Is the decode method doing additional date validation to meet this criterion?

> Clients MUST validate the iat and exp claims in the same way they are validated in ID Tokens.

I assume from the error handling and `leeway` arg it is, but just want to to check.

Reply:
Also, is this view easily unit testable? It would be good to check that the various verification rules are applied as expected

Reply:
have extracted this logic into a service object with unit tests